This patch makes sure that w3m or others will not pollute $HOME/

Avoid pointless bashisms.
Avoid use of non POSIX local builtin.
--- xmlto.in.orig	2015-11-16 14:07:10 UTC
+++ xmlto.in
@@ -78,13 +78,15 @@ EOF
 #  * Remembers the temporary file's name so it can be deleted on exit
 #  * If the failure message is empty or missing, exits on failure
 make_temp () {
-  local dirflag="" prefix="@PACKAGE@"
+  dirflag="" prefix="@PACKAGE@"
   [ "$1" = "-d" ] && { dirflag="-d"; shift; }
   [ -n "$1" ] && prefix="@PACKAGE@-$1"
 
-  if eval $2='$(${MKTEMP} $dirflag "${TMPDIR:-/tmp}/${prefix}.XXXXXX")'
+  tmpnam=$(${MKTEMP} $dirflag "${TMPDIR:-/tmp}/${prefix}.XXXXXX")
+  if [ $? = 0 ]
   then
-    CLEANFILES="$CLEANFILES ${!2}"
+    eval $2="$tmpnam"
+    CLEANFILES="$CLEANFILES $tmpnam"
     return 0
   elif [ -z "$3" ]
   then
@@ -97,7 +99,7 @@ make_temp () {
 }
 
 compute_searchpath () {
-  local oldIFS="${IFS}"
+  oldIFS="${IFS}"
   IFS=":"
   for asearchpath in "$1"; do
     # wrangle relative paths into absolute ones so that the user
@@ -578,7 +580,7 @@ fi
 if [ "$PROFILE" -eq 1 ]
 then
   PROF_PROCESSED="$XSLT_PROCESSED_DIR/$(basename "${INPUT_FILE%.*}").prof"
-  PROFILE_STYLESHEET="http://docbook.sourceforge.net/release/xsl/current/profiling/profile.xsl"
+  PROFILE_STYLESHEET="http://cdn.docbook.org/release/xsl/current/profiling/profile.xsl"
   [ "$VERBOSE" -ge 1 ] && echo >&2 "Profiling stylesheet: ${PROFILE_STYLESHEET}"
 
   [ "${VERBOSE}" -ge 1 ] && \
@@ -651,4 +653,6 @@ then
   # Extremely verbose
   BASH="${BASH} -x"
 fi
+make_temp -d temphome HOME || exit 1
+export HOME
 ${BASH} "$FORMAT" post-process || exit 1
